ste/mgr-JobPartTransferMgr.go

```diff
@@ -270,6 +275,67 @@ func (jptm *jobPartTransferMgr) FileCountLimiter() common.CacheLimiter {
 	return jptm.jobPartMgr.FileCountLimiter()
 }
 
+// WaitUntilLockDestination does two things. It respects any limit that may be in place on the number of
+// active destination files (by blocking until we are under the max count), and it
+// registers the destination as "locked" in our internal map. The reason we
+// lock internally like this is:
+// (a) it is desirable to have some kind of locking because there are (very rare) edge cases where
+// we may map two source files to one destination.  This can happen any time we mutate the destination name
```

Comment:
>we may map two source files to one destination. [](start = 3, length = 47)

Isn't it more common than we think? doesn't windows filesystem ignore casing, so it's possible that many blobs map to the same local file?

Reply:
Oh, yes. That too!!!  Technically, the

Reply:
I've update the comment accordingly.